### ISSUE TITLE
Chat - Accessing an attachment via link that is not found shows the error page twice

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#25779]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___
**Issue found when executing PR** https://github.com/Expensify/App/pull/25516
## Action Performed:
1. Open app New Expensify
2. Log in
3. Open any chat/report
4. Send this link https://staging.new.expensify.com/r/7736083827006365/attachment?source=/chat-attachments/8443634480793859480/w_3de27a40d82c4442386d0447d8dab8af403f786f3.png
5. Tap on the link
6. Tap on the back button

## Expected Result:
The user should be taken to the chat and stay there

## Actual Result:
The error page opens again, so that the user has to close it again

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.56.18

**Reproducible in staging?:** Yes

**Reproducible in production?:** No

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/78819774/df0eea2e-97dc-40c4-b616-b2ac5ec6254f

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause - Internal Team

**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by jeet-dhandha** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
- `Hmm...its not here` - page is coming up twice for an attachment which we do not have access to.

### What is the root cause of that problem?
- The page is coming up first time when we are trying to access the attachment.
- Second time its coming up when we press back button on attachment modal. As its trying to navigate to a reportID which we do not have access to and thus showing `Hmm...its not here` page.

### What changes do you think we should make in order to solve the problem?
- We can update the `dismissModal` function of `Navigation.js` to check if any report is available for given `targetReportID`.
- And also we will have to remove `isLoadingReportActions` key from the found report as it will set key for unknown report in `onyx`.

- Finally, After changes if we press on back it should just close the Attachment Modal.

### What alternative solutions did you explore? (Optional)
- N/A

**Comment by marcaaron** [comment]
@lanitochka17 What platforms is this related to? I can't reproduce on web. Please provide clear reproduction steps.

I don't think this needs to be a blocker. Getting linked to an attachment that doesn't exist seems like a minor flow.

**Comment by jeet-dhandha** [comment]
@marcaaron Try these steps:
1. Open the “Android” app.
2. Send the above link in one of your chat.
3. Press on the link and after that verify the not found page in attachment modal.
4. Press back on top and verify the not found page displaying second time.

**Comment by sobitneupane** [comment]
I will review the proposals today.

**Comment by sobitneupane** [comment]
@joelbettner @MitchExpensify 

We see two error pages here. One is for attachment and one is for the report. We first show the error page for attachment and after user dismisses the modal, user navigates to the report (after split of seconds on original report).

What is the expected output here? IMO, user should be first navigated to the report and to the attachment (In this case, two error pages will be shown) . Or, if user doesn't have access to the report, attachment modal won't be shown at all (In this case, only one error page will be shown).

**Comment by joelbettner** [comment]
@sobitneupane After the user is shown the error for the attachment, we should just navigate back to the chat where the deep link is. Basically, do not show the second error (for the report) and just take the user back to where they were. I think that is what @tienifr was saying, but not totally sure.

**Comment by MitchExpensify** [comment]
Agree that sounds expected to me @joelbettner :

> After the user is shown the error for the attachment, we should just navigate back to the chat where the deep link is. Basically, do not show the second error (for the report) and just take the user back to where they were

**Comment by sobitneupane** [comment]
Thanks for the [update](https://github.com/Expensify/App/issues/25779#issuecomment-1705667724) @joelbettner 

But what should happen after user closes the Attachment Modal if user has access to the report? Should user be navigated back to the chat with the deeplink or chat with the attachment?

**Comment by sobitneupane** [comment]
@MitchExpensify We can take the user back to the chat with deep link. But that's not the behavior now. As of now, we open the attachment and after user closes the attachment, we open the report with the attachment. 

So, I was just looking for confirmation if we want to take user back to chat with deep link even if user has access to the chat with attachment.

**Comment by MitchExpensify** [comment]
Thanks! We want to take the user back to the last screen they were on if they click back @sobitneupane

**Comment by sobitneupane** [comment]
will review the proposals shortly

**Comment by sobitneupane** [comment]
As we have decided to go to the initial chat with deeplink (of attachment) instead of chat with attachment whenever user closes the modal (with either Not Found or attachment), alternative [proposal](https://github.com/Expensify/App/issues/25779#issuecomment-1692975388) from @tienifr does the job.

🎀 👀 🎀 C+ reviewed

**Comment by joelbettner** [comment]
> alternative https://github.com/Expensify/App/issues/25779#issuecomment-1692975388 from @tienifr does the job.

@sobitneupane that sounds good to me. Do I need to assign @tienifr to this issue?

**Comment by sobitneupane** [comment]
Yes, @joelbettner.  He is supposed to create a pull request only after being assigned to the task.

**Comment by MitchExpensify** [comment]
Friendly bump @joelbettner, 🙇‍♂️

**Comment by MitchExpensify** [comment]
Quick Q for ya here @joelbettner https://github.com/Expensify/App/issues/25779#issuecomment-1736589524

**Comment by joelbettner** [comment]
Sorry for the late reply, y'all. I assigned @tienifr to the issue.

**Comment by MitchExpensify** [comment]
I think the bonus is fair if all that delayed us was internal review - Do you agree @joelbettner ?

**Comment by joelbettner** [comment]
@MitchExpensify I agree.

**Comment by MitchExpensify** [comment]
**Payment summary:**

- $1500 @sobitneupane does not require payment (Eligible for Manual Requests)
- $1500 @tienifr requires payment [offer](https://www.upwork.com/nx/wm/offer/27121583) (Contributor)

**Comment by MitchExpensify** [comment]
Paid @tienifr!

**Comment by JmillsExpensify** [comment]
$1,500 payment approved for @sobitneupane based on BZ summary.
